pkg/proxy/nftables/proxier.go

```diff
@@ -1144,25 +1276,32 @@ func (proxier *Proxier) syncProxyRules() {
 		allEndpoints := proxier.endpointsMap[svcName]
 		clusterEndpoints, localEndpoints, allLocallyReachableEndpoints, hasEndpoints := proxy.CategorizeEndpoints(allEndpoints, svcInfo, proxier.nodeLabels)
 
+		// svcTx is used for all service-related chains and their elements.
+		// If no changes were done to the service or its endpoints, these objects may be skipped.
+		svcTx := tx
+		if tryPartialSync && !serviceUpdateResult.UpdatedServices.Has(svcName.NamespacedName) && !endpointUpdateResult.UpdatedServices.Has(svcName.NamespacedName) {
+			svcTx = nil
+		}
```

Comment:
Hm... at this point, `svcTx` _only_ gets used to pass to `ensureChain`. I think it would be clearer to just add a `flush bool` arg to `ensureChain`, which would be `true` in most places, but then here you'd have

    rewriteServiceChains := !tryPartialSync ||
            serviceUpdateResult.UpdatedServices.Has(svcName.NamespacedName) ||
            endpointUpdateResult.UpdatedServices.Has(svcName.NamespacedName)

and then below you'd have things like

    ensureChain(epInfo.chainName, svcTx, activeChains, rewriteServiceChains)

and

    if !rewriteServiceChains {
            continue
    }